decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/data/BasesMask.java

```diff
@@ -0,0 +1,177 @@
+package ca.on.oicr.pde.deciders.data;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ *
+ * @author mlaszloffy
+ */
+public class BasesMask {
+
+    private final Integer readOneIncludeLength;
+    private final Integer readOneIgnoreLength;
+
+    private final Integer indexOneIncludeLength;
+    private final Integer indexOneIgnoreLength;
+
+    private final Integer indexTwoIncludeLength;
+    private final Integer indexTwoIgnoreLength;
+
+    private final Integer readTwoIncludeLength;
+    private final Integer readTwoIgnoreLength;
+
+    public BasesMask(Integer readOneIncludeLength, Integer readOneIgnoreLength, Integer indexOneIncludeLength, Integer indexOneIgnoreLength, Integer indexTwoIncludeLength, Integer indexTwoIgnoreLength, Integer readTwoIncludeLength, Integer readTwoIgnoreLength) {
+        this.readOneIncludeLength = readOneIncludeLength;
+        this.readOneIgnoreLength = readOneIgnoreLength;
+        this.indexOneIncludeLength = indexOneIncludeLength;
+        this.indexOneIgnoreLength = indexOneIgnoreLength;
+        this.indexTwoIncludeLength = indexTwoIncludeLength;
+        this.indexTwoIgnoreLength = indexTwoIgnoreLength;
+        this.readTwoIncludeLength = readTwoIncludeLength;
+        this.readTwoIgnoreLength = readTwoIgnoreLength;
+    }
+
+    public BasesMask(String readOneIncludeLength, String readOneIgnoreLength, String indexOneIncludeLength, String indexOneIgnoreLength, String indexTwoIncludeLength, String indexTwoIgnoreLength, String readTwoIncludeLength, String readTwoIgnoreLength) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)